glommio/src/sys/uring.rs

```diff
@@ -34,6 +35,8 @@ use uring_sys::IoRingOp;
 
 use super::{EnqueuedSource, TimeSpec64};
 
+const MSG_ZEROCOPY: i32 = 0x4000000;
```

Comment:
Does this flag apply in both these situations (send/recv)? https://www.kernel.org/doc/html/v4.15/networking/msg_zerocopy.html . It seems like it applies to sends.. doesn't say anything about recv's.

Reply:
you are actually right. Good call, I had read that same page that you linked but completely missed that point

Reply:
this has big implications though - so thanks a lot for catching it early.
the Zerocopy flag was my main reason to use the DmaBuffer infrastructure for reads. Because those pages are already pinned and registered, I figured we could gain some cycles here (although I haven't validated that)

It does make it harder to use the uring buffer selection infrastructure, so I made the trade off.
I may not do buffer selection now - but I would definitely like to make it a possibility if there is nothing to gain from the zerocopy flag.

So I will likely rework this to work with standard buffers.

We may also try to add a yolo-path to fill_buff as well (see the comments in https://github.com/DataDog/glommio/issues/207 from the reporter).

All in all this will give me some headache, but a good headache. So don't feel bad =)

Reply:
update: won't implement yolo for the fill_buf path. It is potentially wasteful because you don't know how many bytes in advance to read and may impose an extra copy since we can't use io_uring buffers for that.

Buffer selection is a bit weird and I am still fighting it. Details here:

https://twitter.com/glcst/status/1337563370740985857

I will merge without it and we can come back to that later.

Reply:
Interesting issue. It looks like you've gone to the right people on twitter, though. The original creator of io_uring is a good person to ask.